rsyslog/go/src/github.com/mmexternal/undefined_field/undefined_field.go

```diff
@@ -0,0 +1,239 @@
+package main
```

Comment:
@richm , is it ok to place the plugin in this path `rsyslog/go/src/github.com/mmexternal/undefined_field`?   Or maybe I'd better map to the rsyslog source tree structure, something like `rsyslog/go/src/github.com/rsyslog/plugins/external/undefined_field/undefined_field.go`?

And what "package" name I'm supposed to use?  Or it does not matter since it's just an independent plugin?

Reply:
> @richm , is it ok to place the plugin in this path `rsyslog/go/src/github.com/mmexternal/undefined_field`? Or maybe I'd better map to the rsyslog source tree structure, something like `rsyslog/go/src/github.com/rsyslog/plugins/external/undefined_field/undefined_field.go`?

Well, the path is supposed to represent the source of the code - so the path `go/src/github.com/mmexternal/undefined_field` would represent `https://github.com/mmexternal/undefined_field` which does not exist.  Which means it would logically go under `go/src/github.com/openshift/origin-aggregated-logging/rsyslog`, but that is redundant.

Therefore, I think the code should live in the `rsyslog` directory.

> 
> And what "package" name I'm supposed to use? Or it does not matter since it's just an independent plugin?

https://golang.org/doc/code.html#PackageNames
"Executable commands must always use package `main`."

This is a standalone executable command, so, `main` is good.

Reply:
Thank you, @richm!  Let me adjust the path to rsyslog including the Dockerfile below.